src/resources/kubernetesRedis.ts

```diff
@@ -0,0 +1,135 @@
+import * as pulumi from '@pulumi/pulumi';
+import * as k8s from '@pulumi/kubernetes';
+import { urnPrefix } from '../constants';
+import { AdhocEnv } from '../utils';
+
+type Image = {
+  repository: pulumi.Input<string>;
+  tag: pulumi.Input<string>;
+};
+
+type Tolerations = {
+  key: pulumi.Input<string>;
+  value: pulumi.Input<string>;
+  operator: pulumi.Input<'Exists' | 'Equal'>;
+  effect: pulumi.Input<'NoExecute' | 'NoSchedule' | 'PreferNoSchedule'>;
+};
+
+type NodeAffinity = {
+  type: pulumi.Input<'soft' | 'hard'>;
+  key: pulumi.Input<string>;
+  values: pulumi.Input<string[]>;
+};
+
+export type K8sRedisArgs = AdhocEnv & {
+  replicas: pulumi.Input<number>;
+  memorySizeGb: pulumi.Input<number>;
+  namespace: pulumi.Input<string>;
+  architecture?: pulumi.Input<'standalone' | 'replication'>;
+  image?: Image;
+  persistence?: pulumi.Input<{
+    enabled?: pulumi.Input<boolean>;
+    [key: string]: unknown;
+  }>;
+  tolerations?: pulumi.Input<{
+    master?: Tolerations;
+    replicas?: Tolerations;
+  }>;
+  nodeSelector?: pulumi.Input<{
+    master?: NodeAffinity;
+    replicas?: NodeAffinity;
+  }>;
+  authKey?: pulumi.Input<string>;
+};
+
+const commonConfiguration = `
+loadmodule /opt/redis-stack/lib/redisearch.so
+loadmodule /opt/redis-stack/lib/rejson.so
+`;
+
+const defaultImage = {
+  repository: 'redis/redis-stack-server',
+  tag: '7.2.0-v10',
+};
+
+const chartVersion = '19.0.2';
+
+export class KubernetesRedis extends pulumi.ComponentResource {
+  constructor(
+    name: string,
+    args: K8sRedisArgs,
+    opts?: pulumi.CustomResourceOptions,
+  ) {
+    super(`${urnPrefix}:KubernetesRedis`, name, args, opts);
+
+    // const tolerations = args.tolerations || [];
+    const persistence = {
+      ...args.persistence,
+      size: pulumi
+        .all([args.memorySizeGb])
+        // Set persistence size to 2x memory size
+        .apply(([memorySizeGb]) => `${Math.max(memorySizeGb * 2)}Gi`),
+    };
+
+    const resources = args.isAdhocEnv
+      ? undefined
+      : {
+          requests: {
+            cpu: '1000m',
```

Comment:
This means one core and redis is single threaded, this is why we don't need it to be customized?

Reply:
Well, it's not that straight forward. Resources (of CPU) is more an allocation CPU time. Probably can make it configurable, then we can tweak it later if we see that `1` is overkill or not (really depends on workloads).